lib/adapter/cucumber/legacy.js

```diff
@@ -120,14 +121,21 @@ const createTestomatFormatter = apiKey => {
         message = 'Undefined steps. Implement missing steps and rerun this scenario';
       }
       console.log(cliMessage);
-      if (status !== TRConstants.PASSED && status !== TRConstants.SKIPPED) {
-        this.status = TRConstants.FAILED;
+      if (status !== STATUS.PASSED && status !== STATUS.SKIPPED) {
+        this.status = STATUS.FAILED;
       }
-      this.client.addTestRun(testId, status, {
-        error: event.result.exception,
+      let error;
+      if (event.result.exception instanceof Error) {
+        error = event.result.exception;
+      }
+
+      this.client?.addTestRun(status, {
+        error,
```

Comment:
I would prefer something like
`error: event.result.exception instanceof Error ? event.result.exception : undefined,`
or
`error: event.result.exception instanceof Error && event.result.exception,`


and remove the code:
```
let error;
if (event.result.exception instanceof Error) {
  error = event.result.exception;
}
```